libraries/freertos_plus/standard/freertos_plus_tcp/source/FreeRTOS_DNS.c

```diff
@@ -1221,7 +1221,7 @@ BaseType_t xDoStore = xExpected;
 #endif /* ipconfigUSE_LLMNR == 1 */
 	} while( 0 );
 
-	if( xExpected != pdFALSE )
+	if( xExpected == pdFALSE )
 	{
 		/* Do not return a valid IP-address in case the reply was not expected. */
 		ulIPAddress = 0uL;
```

Comment:
About the commit "Ignore replies that are not expected" : the comment was correct but the test was not:

    - if( xExpected != pdFALSE )
    + if( xExpected == pdFALSE )
      {
          /* Do not return a valid IP-address in case the reply was not expected. */
          ulIPAddress = 0uL;
      }
      return ulIPAddress;

While testing #861, I noticed that the synchronous look-up didn't work, due to the above typo.
After the above change, I re-tested FreeRTOS_DNS with the PC demo project and with an Atmel project, both synchronous and asynchronous.


What I did find is that LLMNR also needs the function `ulDNSHandlePacket()` :

    -    #if( ipconfigDNS_USE_CALLBACKS == 1 )
    +    #if( ipconfigDNS_USE_CALLBACKS == 1 ) || ( ipconfigUSE_LLMNR == 1 )
    uint32_t ulDNSHandlePacket( NetworkBufferDescriptor_t *pxNetworkBuffer )